code/modules/mapping/mapping_helpers.dm

```diff
@@ -546,6 +546,27 @@
 		log_mapping("[src] at [AREACOORD(src)] [(apc_area.type)] tried to set [target]'s charge to 100 but it's already at 100!")
 	target.full_charge = TRUE
 
+//Used to turn off lights in areas.
+/obj/effect/mapping_helpers/turn_off_lights
+	name = "area lights off helper"
+	icon_state = "lights_off"
+	late = TRUE
+
+/obj/effect/mapping_helpers/turn_off_lights/Initialize(mapload)
+	. = ..()
+	if(!mapload)
+		log_mapping("[src] spawned outside of mapload!")
+		return INITIALIZE_HINT_QDEL
+	payload()
+	qdel(src)
+	return INITIALIZE_HINT_LATELOAD
+
+/obj/effect/mapping_helpers/turn_off_lights/proc/payload()
```

Comment:
you should name this something that better describes it

Reply:
`turn_off_lights_in_area` or something? no idea tbh

Reply:
The proc I mean, sorry shoulda been more clear the type was named fine

Reply:
This is fine.